init-tools.sh

```diff
@@ -109,6 +110,7 @@ if [ ! -e $__INIT_TOOLS_DONE_MARKER ]; then
 
     Tools/crossgen.sh $__scriptpath/Tools
 
+    mkdir -p $__INIT_TOOLS_DONE_MARKER_DIR
     touch $__INIT_TOOLS_DONE_MARKER
```

Comment:
touch just silently fails if the directory doesn't exist? Or are we just ignoring the failure?